src/ens/records.c

```diff
@@ -0,0 +1,164 @@
+#include <string.h>
+
+#include "../covid_types.h"
+#include "records.h"
+#include "sequencenumber.h"
+#include "storage.h"
+
+int ens_records_iterator_init_range(record_iterator_t* iterator,
+                                    record_sequence_number_t* opt_start,
+                                    record_sequence_number_t* opt_end) {
+    iterator->sn_next = opt_start ? *opt_start : get_oldest_sequence_number();
+    iterator->sn_end = opt_end ? *opt_end : get_latest_sequence_number();
+    if (get_num_contacts() == 0) {
+        iterator->finished = true;  // no contacts -> no iteration :)
+    }
```

Comment:
We need to set finished to false in case the underlying memory has been previously used as an iterator.
```suggestion
    }
    else {
        iterator->finished = false;
    
    }
```